docs/decisions/0002-lerna-monorepo.rst

```diff
@@ -0,0 +1,108 @@
+
+Before further expanding the scope of what ``frontend-enterprise`` entails (i.e., adding UI components related to enterprise logistration), we should invest in splitting the domains/components of ``frontend-enterprise`` out into separate NPM packages to have better separation of concerns for consuming libraries and micro-frontend applications, reducing confusion and errors around dependencies and peer dependencies.
+
+********
+Decision
+********
+
+By separating the components of ``frontend-enterprise`` into different NPM packages, we are introducing some complexity in terms of the set up of this repository. Rather than split each domain into its own Github repository, we are taking the approach of keeping each domain in the same repository as a monorepo managed by `Lerna <https://lerna.js.org/>`_.
+
+The 3 initial packages contained in the refactored Lerna monorepo are as follows:
+
+1. ``@edx/frontend-enterprise-catalog-search``
+2. ``@edx/frontend-enterprise-logistration``
+3. ``@edx/frontend-enterprise-utils``
+
+By splitting ``frontend-enterprise`` into separate NPM packages, only the dependencies relevant to each individual package are included in the package.json file. This solves the issue of requiring peer dependencies for consuming micro-frontends to install ``frontend-enterprise`` when those peer dependencies (e.g., react-instantsearch-dom) are not relevant. While the consuming libraries and micro-frontends were already importing only the necessary code thanks to tree shaking, we can now also create a proper separation of concerns for dependencies in package.json.
+
+================================
+Packages
+================================
+
+--------------------------------
+``@edx/frontend-enterprise-catalog-search``
+--------------------------------
+
+This package exports UI components, React hooks, and actions functions related to Enterprise catalog search using `Algolia <algolia.com>`_ as a hosted search backend. This package will be used wherever it's
+necessary to allow users to search the catalog(s) available to them through their Enterprise Customer.
+
+--------------------------------
+``@edx/frontend-enterprise-logistration``
+--------------------------------
+
+This package exports a UI component related to logistration for Enterprise users. This package will be helpful for Enterprise micro-frontends to share logic related to redirecting enterprise users to the logistration page branded with their Enterprise Customer's logo and additional messaging.
+
+--------------------------------
+``@edx/frontend-enterprise-utils``
```

Comment:
what is the likelihood this becomes a catch-all package?

Ie I have a new shared component/function/thing and maybe it's not explicitly or obviously related to catalog search or logistration? I just imagine this package could eventually run into the same issues the overall repo did (too big, too many irrelevant dependencies, etc.) and why we split it into 3 packages. Do we plan to have any guidelines on that? Or is that a decision we make at some point if we see it getting too complicated?

Reply:
Good call out, @taliaedX! We definitely run the risk of it becoming a catch-all package and start running into similar issues. I agree we should add more explicit guidance around this somewhere, perhaps a section in the README in the root of the monorepo that provides some rough guidelines / considerations to make on whether to add new things to existing packages or start a new package (and how to do so).

I might bucket that into the next follow-up PR when updating some of the READMEs anyways?